handlers/index.js

```diff
@@ -0,0 +1,199 @@
+        redirectUri: authCallback,
+        state: session.state,
+        access_mode: "online", // Always generate online mode token for extension launch
+    });
+    await SessionStorage.saveSession(session);
+    logger.debug(`Redirecting after install callback to url: ${redirectUrl}`);
+    return {
+        redirectUrl: redirectUrl,
+        fdkSession: session,
+    };
+};
+
+const fpAuth = async (reqObj, state, code, ext, sessionId) => {
+    const fdkSession = await SessionStorage.getSession(sessionId);
+    if (!fdkSession) { 
+        throw new FdkSessionNotFoundError(
+            "Can not complete oauth process as session not found"
+        );
+    }
+    if (fdkSession.state !== state) {
+        throw new FdkInvalidOAuthError("Invalid oauth call");
+    }
+    const companyId = fdkSession.company_id;
+
+    const platformConfig = ext.getPlatformConfig(fdkSession.company_id);
+    await platformConfig.oauthClient.verifyCallback({code});
+
+    let token = platformConfig.oauthClient.raw_token;
+    let sessionExpires = new Date(Date.now() + token.expires_in * 1000);
+
+    fdkSession.expires = sessionExpires;
+    token.access_token_validity = sessionExpires.getTime();
+    fdkSession.updateToken(token);
+
+    await SessionStorage.saveSession(fdkSession);
+
+    // Generate separate access token for offline mode
+    if (!ext.isOnlineAccessMode()) {
+        let sid = Session.generateSessionId(false, {
+            cluster: ext.cluster,
+            companyId: companyId,
+        });
+        let session = await SessionStorage.getSession(sid);
+        if (!session) {
+            session = new Session(sid);
+        } else if (session.extension_id !== ext.api_key) {
+            session = new Session(sid);
+        }
+
+        let offlineTokenRes =
+            await platformConfig.oauthClient.getOfflineAccessToken(
+                ext.scopes,
+                code
+            );
+
+        session.company_id = companyId;
+        session.scope = ext.scopes;
+        session.state = fdkSession.state;
+        session.extension_id = ext.api_key;
+        offlineTokenRes.access_token_validity =
+            platformConfig.oauthClient.token_expires_at;
+        offlineTokenRes.access_mode = "offline";
+        session.updateToken(offlineTokenRes);
+        await SessionStorage.saveSession(session);
+    }
+    
+    if (
+        ext.webhookRegistry.isInitialized &&
+        ext.webhookRegistry.isSubscribeOnInstall
+    ) {
+        const client = await ext.getPlatformClient(companyId, fdkSession);
+        await ext.webhookRegistry.syncEvents(client, null, true).catch((err) => {
+            logger.error(err);
+        });
+    }
+
+    let redirectUrl = await ext.callbacks.auth(reqObj);
+    logger.debug(`Redirecting after auth callback to url: ${redirectUrl}`);
+    return {
+        redirectUrl: redirectUrl,
+        fdkSession: fdkSession,
+    };
+};
+
+const fpAutoInstall = async (reqObj, company_id, code, ext) => {
```

Comment:
Add named function here as it will be helpful while reading stack trace.  @jinalvirani

Reply:
Resolved